data_pipelines/assets/flood/discharge_new.py

```diff
@@ -0,0 +1,327 @@
+import os
+from datetime import datetime, timedelta
+import pandas as pd
+
+import xarray as xr
+from dagster import AssetExecutionContext, asset
+import dask.dataframe as dd
+
+from data_pipelines.resources.dask_resource import DaskResource
+from data_pipelines.resources.glofas_resource import CDSClient, CDSConfig
+from data_pipelines.utils.flood.config import *
+from data_pipelines.utils.flood.etl.filter_by_upstream import apply_upstream_threshold
+from data_pipelines.utils.flood.etl.raster_converter import RasterConverter
+from data_pipelines.utils.flood.etl.transforms import (
+    compute_flood_intensity,
+    compute_flood_peak_timing,
+    compute_flood_tendency,
+    compute_flood_threshold_percentages,
+    add_geometry,
+)
+from data_pipelines.utils.flood.etl.utils import restrict_dataset_area
+
+from data_pipelines.partitions import discharge_partitions
+
+
+def make_path(*args) -> str:
+    path = os.path.join(*args)
+    os.makedirs(os.path.dirname(path), exist_ok=True)
+    return path
+
+
+# This is the partitioned version of the raw_discharge asset
+# It performs prallelized queries of the CDS API for each leadtime hour
+# Such parallelization can cause the pipeline to fail with no clear error message
+@asset(
+    key_prefix=["flood"],
+    partitions_def=discharge_partitions,
+    io_manager_key="grib_io_manager",
+    # op_tags={"dagster/concurrency_key": "conc_test"}, # this doesn't work
+)
+def raw_discharge(context: AssetExecutionContext, client: CDSClient) -> None:
+    date_for_request = datetime.utcnow()  # - timedelta(days=TIMEDELTA)
+
+    query_buffer = GLOFAS_RESOLUTION * GLOFAS_BUFFER_MULT
+    lat_min = GLOFAS_ROI_CENTRAL_AFRICA["lat_min"]
+    lat_max = GLOFAS_ROI_CENTRAL_AFRICA["lat_max"]
+    lon_min = GLOFAS_ROI_CENTRAL_AFRICA["lon_min"]
+    lon_max = GLOFAS_ROI_CENTRAL_AFRICA["lon_max"]
+
+    area = [
+        lat_max + query_buffer,
+        lon_min - query_buffer,
+        lat_min - query_buffer,
+        lon_max + query_buffer,
+    ]
+
+    if USE_CONTROL_MEMBER_IN_ENSEMBLE:
+        product_type = ["control_forecast", "ensemble_perturbed_forecasts"]
+        print("Retrieving both control and ensemble")
+    else:
+        product_type = "ensemble_perturbed_forecasts"
+        print("Retrieving only ensemble")
+
+    l_hour = context.partition_key
+    target_file_path = make_path(
+        OPENEPI_BASE_PATH,
+        *context.asset_key.path,
+        f"{l_hour}.grib",
+    )
+    os.makedirs(os.path.dirname(target_file_path), exist_ok=True)
+
+    # Define the config
+    config = CDSConfig(
+        year=date_for_request.year,
+        month=date_for_request.month,
+        day=date_for_request.day,
+        leadtime_hour=l_hour,
+        area=area,
+        product_type=product_type,
+    )
+
+    # Convert config to a dictionary
+    request_params = config.to_dict()
```

Comment:
There are some additional default params within the CDSConfig class

Reply:
As far as I can tell the `CDSConfig` class is only used here, in which case I still think it would clearer to just construct the dictionary directly. Do you plan on adding other assets where it will be used?

Reply:
True, a simple dictionary is better here